streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -312,16 +305,20 @@ private void addToExceptionsAndFailedTasksThenClearUpdatingTasks(final Exception
             updatingTasks.clear();
         }
 
-        private void waitIfAllChangelogsCompletelyRead() throws InterruptedException {
-            if (isRunning.get() && changelogReader.allChangelogsCompleted()) {
-                tasksAndActionsLock.lock();
-                try {
-                    while (tasksAndActions.isEmpty() && !isTopologyResumed.get()) {
-                        tasksAndActionsCondition.await();
-                    }
-                } finally {
-                    tasksAndActionsLock.unlock();
+        private void waitIfAllChangelogsCompletelyRead() {
+            tasksAndActionsLock.lock();
```

Comment:
Nit: the name `tasksAndActionsLock/Condition` is probably not very adequate anymore. But we don't need to change it in this PR.

Reply:
The name comes from the `TaskAndAction` class, which I think is still appropriate even if we are left with `add` and `remove` actions. What's your suggestion for a better naming?

Reply:
I re-read the code and could not came up with a better name. Let's keep it as it is for now.